services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/store/device/TableAdapterStore.java

```diff
@@ -114,4 +195,52 @@ public TableAdapterStore(final SQLClient client, final Tracer tracer, final Stat
 
     }
 
+    /**
+     * Resolve a list of group members.
+     *
+     * @param tenantId The tenant the device belongs to.
+     * @param viaGroups The viaGroups list of a device. This list contains the ids of groups.
+     * @param spanContext The span to contribute to.
+     *
+     * @return A future tracking the outcome of the operation.
+     *
+     * @see org.eclipse.hono.deviceregistry.service.device.AbstractRegistrationService#resolveGroupMembers(String, JsonArray, Span)
+     */
+    public Future<Set<String>> resolveGroupMembers(final String tenantId, final Set<String> viaGroups, final SpanContext spanContext) {
+
+        final Span span = TracingHelper.buildChildSpan(this.tracer, spanContext, "resolve group members", getClass().getSimpleName())
+                .withTag("tenant_instance_id", tenantId)
```

Comment:
Could you have a look at the comment in the old PR (https://github.com/eclipse/hono/pull/2118#discussion_r482336686) and elaborate a bit on the internal tenant id you mentioned?
Do you mean the `TenantKey#name` value (javadoc: "tenant name in an external system")? 

To me, the `resolveGroupMembers` method above doesn't seem to be invoked in a way that such a name/identifier is used for the first parameter here, though.

Reply:
Taking a closer look at the code, I think we have a problem here with the way this was migrated from EnMasse.

The "name" used to be the user facing name, where "id" is the internal ID, e.g. in the database. The tenant information service can (if implemented that way) translate from the external name to the internal name.

The JDBC device registry must only use the internal name.

So the `resolveGroupMembers` method, should either receive the internal ID (which is the current code), receive a `TenantKey` and use the internal ID, or do the lookup (via the tenant information service) itself.

I think it would be ok to ignore this for the moment, and simply use the "ID" as "tenant_id". And sort that out later.